x/epochs/spec/README.md

```diff
@@ -0,0 +1,45 @@
+<!--
+order: 0
+title: "Epochs Overview"
+parent:
+  title: "epochs"
+-->
+
+# `epochs`
+
+## Abstract
+
+The purpose of `epochs` module is to provide generalized epoch interface to other modules so that they can easily implement epochs without keeping own code for epochs.
+
+## Implementation
+
+### Keeper functions
+```go
+// Keeper is the interface for lockup module keeper
+type Keeper interface {
+  // GetEpochInfo returns epoch info by identifier
+  GetEpochInfo(ctx sdk.Context, identifier string) types.EpochInfo
+  // SetEpochInfo set epoch info
+  SetEpochInfo(ctx sdk.Context, epoch types.EpochInfo) 
+  // DeleteEpochInfo delete epoch info
+  DeleteEpochInfo(ctx sdk.Context, identifier string)
+  // IterateEpochInfo iterate through epochs
+  IterateEpochInfo(ctx sdk.Context, fn func(index int64, epochInfo types.EpochInfo) (stop bool))
+  // Get all epoch infos
+  AllEpochInfos(ctx sdk.Context) []types.EpochInfo
+}
+```
+
+### Hooks
+```go
+  // the first block whose timestamp is after the duration is counted as the end of the epoch
+  AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64)
+  // new epoch is next block of epoch end block
+  BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64)
+```
+
+### How modules receive hooks
```

Comment:
Can there be some comments as to what happens if an epoch is really short? In the code it says that epochs must last at least two blocks. If its complex to reason about, I think its fine to say that "we require, but do not enforce in code, that all epoch lengths are much longer than two block times", and leave it as an issue for us to finish reasoning about later

Reply:
Added description on this

Reply:
Thanks